substrate-node/pallets/pallet-tfgrid/src/node.rs

```diff
@@ -227,3 +227,161 @@ fn validate_serial_number(input: &[u8]) -> bool {
         .iter()
         .all(|c| matches!(c, b'a'..=b'z' | b'A'..=b'Z' | b'0'..=b'9' | b'-' | b'_'))
 }
+
+pub const ONE_THOUSAND: u128 = 1_000;
+pub const GIB: u128 = 1024 * 1024 * 1024;
+
+/// A resources capacity that countains HRU, SRU, CRU and MRU in integer values.
+#[derive(Encode, Decode, Eq, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+#[scale_info(skip_type_params(T))]
+#[codec(mel_bound())]
+pub struct Resources<T: Config> {
+    pub hru: u64,
+    pub sru: u64,
+    pub cru: u64,
+    pub mru: u64,
+    _marker: PhantomData<T>,
+}
+
+pub trait Capacity {
```

Comment:
Is there a way to avoid creating this `Capacity` trait?

Reply:
yes, just make `implementations` on the `Resources` type. See: https://github.com/threefoldtech/tfchain/blob/799c8bfba6dacf11fb53fe59cd9173070178c23c/substrate-node/support/src/types.rs#L135

Can you also rework: https://github.com/threefoldtech/tfchain/blob/development/substrate-node/support/src/resources.rs

Also make it an implementation of `Resources`.


I suggest you move the type definition of `Resources` to `support/src/resources.rs` and make all the implementations there.

Reply:
Done!